### ISSUE TITLE
False positive on `OutputInterface::write()` since 1.2.6

### ISSUE THREAD

**Issue by NoiseByNorthwest** [phpstan/phpstan-symfony#295]
Since the upgrading of phpstan-symfony from 1.2.5 to 1.2.6 I have the following error when running PHPStan. It really looks like a false positive.

```
Parameter #3 $options of method                                        
         Symfony\Component\Console\Output\OutputInterface::write() expects      
         0|1|2|3|4|5|6|7|16|17|18|19|20|21|22|23|32|33|34|35|36|37|38|39|48|49  
         |50|51|52|53|54|55|64|65|66|67|68|69|70|71|80|81|82|83|84|85|86|87|96  
         |97|98|99|100|101|102|103|112|113|114|115|116|117|118|119|128|129|130  
         |131|132|133|134|135|144|145|146|147|148|149|150|151|160|161|162|163|  
         164|165|166|167|176|177|178|179|180|181|182|183|192|193|194|195|196|1  
         97|198|199|208|209|210|211|212|213|214|215|224|225|226|227|228|229|23  
         0|231|240|241|242|243|244|245|246|247|256|257|258|259|260|261|262|263  
         |272|273|274|275|276|277|278|279|288|289|290|291|292|293|294|295|304|  
         305|306|307|308|309|310|311|320|321|322|323|324|325|326|327|336|337|3  
         38|339|340|341|342|343|352|353|354|355|356|357|358|359|368|369|370|37  
         1|372|373|374|375|384|385|386|387|388|389|390|391|400|401|402|403|404  
         |405|406|407|416|417|418|419|420|421|422|423|432|433|434|435|436|437|  
         438|439|448|449|450|451|452|453|454|455|464|465|466|467|468|469|470|4  
         71|480|481|482|483|484|485|486|487|496|497|498|499|500|501|502|503,    
         int given.                                                             
 ------ ----------------------------------------------------------------------- 
```

**Comment by ondrejmirtes** [comment]
This is the reason: https://github.com/phpstan/phpstan-symfony/pull/293

**Comment by ondrejmirtes** [comment]
What piece of code reports the error for you?

**Comment by NoiseByNorthwest** [comment]
This is a method which wraps the call to `OutputInterface::write()` and accepts an "int $options" which is thus wider (int)  than the accepted type provided by the `OutputInterface::write()`'s stub (int-mask-of<...>).

```php
private function write(
        string $s,
        bool $newLine = true,
        int $options = 0
    ): void {
        $this
            ->executionContext
            ->getOutput()
            ->write($s, $newLine, $options)
        ;
    }
```

**Comment by NoiseByNorthwest** [comment]
Fixed with

```php
/**
     * @param string                                                              $s
     * @param bool                                                                $newLine
     * @param int-mask-of<OutputInterface::VERBOSITY_*|OutputInterface::OUTPUT_*> $options
     */
    private function write(
        string $s,
        bool $newLine = true,
        int $options = 0
    ): void {
        $this
            ->executionContext
            ->getOutput()
            ->write($s, $newLine, $options)
        ;
    }
```

**Comment by NoiseByNorthwest** [comment]
If you think this is the way to do I'll close this issue.

**Comment by ondrejmirtes** [comment]
Yes :)
